awscli/clidriver.py

```diff
@@ -76,12 +77,25 @@ def _set_user_agent_for_session(session):
     session.user_agent_extra = 'botocore/%s' % botocore_version
 
 
+def _update_config_chain(session):
```

Comment:
Talked offline about this but let's just make this an internal method and pull in the logic for configuration into this method instead of having the `awscli.configuration` module.